artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/management/CoreNotificationType.java

```diff
@@ -39,7 +39,15 @@
    PROPOSAL(18),
    PROPOSAL_RESPONSE(19),
    UNPROPOSAL(20),
-   CONSUMER_SLOW(21);
+   CONSUMER_SLOW(21),
+   ADDRESS_ADDED(22),
+   ADDRESS_REMOVED(23),
+   CONNECTION_CREATED(24),
+   CONNECTION_DESTROYED(25),
+   SESSION_CREATED(26),
+   SESSION_CLOSED(27),
+   MESSAGE_DELIVERED(28),
```

Comment:
isn't that too costly to the broker?

that's a lot of notifications here. I don't think we should do on MESSAGE_DELIVERED.

Reply:
No because every notification is off by default.  The user can configure what they want.  By default if you configure the plugin it doesn't do anything unless the user sets a type to true to send.  So if someone doesn't care about message delivered notifications then just leave that flag as false.

Reply:
@cshannon when  you go to ServerConsumerImpl, there's already a previous call you added to afterDeliver, that's apparently gets called every time.


Shouldn't that be changed to call this new component you wrote?


I guess that's where I'm getting confused. I read the code as part of your change now and I see that's an older change you added back in Jan this year.


Can you take a look for the relationship between that change and this one now?

Reply:
@cshannon this one: https://github.com/apache/activemq-artemis/blob/39fc8cf1413c0f3cbb88913531753e8a7dba3d07/artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java#L437